src/DistortableImageOverlay.js

```diff
@@ -81,6 +89,11 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
     }
 
     this.fire('add');
+
+    // SEGUN ------------------------------------------------------------------------------------------------------------------------------
+    L.DomEvent.on(this.getElement(), 'mouseover mousemove', this._activateTooltip, this);
```

Comment:
So I think this is getting activated separately and in parallel for every image. Is there a way we could move this code to be at the level of the whole map instead of just every image, and then we could centralize it? I think we would have to centralize the event management but then also still check each image to see if we're hovering over it?

I'm not sure but definitely we are firing mouse move from multiple listeners because this is fired from each image. What do you think?

It looks great so far!!! Tough problem!

Reply:
Yes, you are right, I intentionally coded the event to fire for each of the imageOverlays to enable me decide the correct tooltip to display. But shouldn't this be the appropriate behaviour since every imageOverlay is an independent object (sort of) with its unique set of properties (including tooltip), event listeners etc. Also, note, multiple sets of mousemove event will only be fired if there are multiple imageOverlays on the tiled layer. 

I don't see how centralizing the said code will not disrupt this behaviour.

Reply:
It seems like the advantage of coding the event listener in the images is that we can access the image title and such. But the disadvantage is that we can't easily coordinate between separate images both reacting. Either we react individually from each image, but then try to check which image should take priority, or we react a single time from the map, and then try to see which image is directly under the mouse, and at the top. So I think your approach does make a lot of sense... I'm just not sure how to also get the information about which image is on top. 

Is `mousemove` limited to when the mouse is over the element that's being listened to? That is, are we getting `mousemove` events from both images? Or are we in a situation where image A has inserted its tooltip, and then image B does it too, but not at the same time? Would it help to have each clear the tooltip first, before adding its own? Or would that flicker back and forth between the two? Maybe that's worth a quick test? 

Sorry I'm not really sure what the answer is, but I think we're getting closer to understanding what's actually happening, and then we can make our best decision. Thanks @segun-codes !!

Reply:
1. The mousemove event is fired in turns, no two imageOverlays can fire the event simultaneously since the mouse can only be placed on one imageOverlay per time.

2. Yes, mousemove event is limited to when the mouse is over the element (i.e., an image overlay) that's being listened to.

3. Tooltip is wired-up once imageOverlay is placed on the tiled layer but the tooltip is only visible when user places or moves cursor over the imageOverlay.

4. So if 10imageOverlays are placed on the tiled layer, the 10 will have their tooltips wired-up.

5. The tooltip is only stripped down (clear, unbound from image) once the imageOverlay is removed from the tiled layer. 

6. Would it help to have each clear the tooltip first = this will be difficult, because the tooltip must be setup as soon an instance of imageOverlay is created otherwise it becomes complicated to tell what tooltip belongs to which image.

8. Since what we are doing is merely adding one more property (i.e., tooltip) to each instance of imageOverlay and simply extending the use of the mousemove event fired by the imageOverlay, I don't think this should obstruct how we currently determine what image is on top.

What do you think?

Many thanks for the support so far @jywarren.

Reply:
What about -- when do we /create/ the tooltip vs. when do we hide or show it? 

And - should mousemove fire only for the top image, or for any image even if another image is covering it?

For clearing the tooltip, I was thinking just hiding it, not deleting it. But this depends on when we create/destroy vs. hide/show it.

Thanks, this is very helpful for diagnosing! Did your more recent changes improve things?

Reply:
Mousemove should fire only for the top image and this is the present behaviour. 

As per clearing tooltip, the tooltip is created when the **place on map** button on the sidebar is clicked. However, the tooltip will remain hidden until mousemove event is fired at which point it becomes visible but then it goes hidden again the moment mouseout event is triggered.  

The tooltip is destroyed when the affected `imageOverlay` is removed from the tiled layer. This behaviour seems logical to me.

As per improvement, none yet. There's a need to manually develop a simple mathematical model to create the effect where the leaflet tooltip follows the mouse. I'll be happy to try this later on.